### ISSUE TITLE
NVIDIA driver too old error

### ISSUE THREAD

**Issue by YuvalNirkin** [pytorch/pytorch#4546]
I am getting the following error:

  File "/home/yuval/Dev/Python/Projects/nlm_cnn/train.py", line 116, in main
    train_loss, train_acc = train(train_loader, model, criterion, optimizer, epoch, cuda, log_writer)
  File "/home/yuval/Dev/Python/Projects/nlm_cnn/train.py", line 147, in train
    for i, (image, target) in enumerate(pbar):
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/tqdm/_tqdm.py", line 953, in __iter__
    for obj in iterable:
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 210, in __next__
    return self._process_next_batch(batch)
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 230, in _process_next_batch
    raise batch.exc_type(batch.exc_msg)
AssertionError: Traceback (most recent call last):
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 64, in _pin_memory_loop
    batch = pin_memory_batch(batch)
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 132, in pin_memory_batch
    return [pin_memory_batch(sample) for sample in batch]
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 132, in <listcomp>
    return [pin_memory_batch(sample) for sample in batch]
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 126, in pin_memory_batch
    return batch.pin_memory()
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/tensor.py", line 82, in pin_memory
    return type(self)().set_(storage.pin_memory()).view_as(self)
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/storage.py", line 83, in pin_memory
    allocator = torch.cuda._host_allocator()
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/cuda/__init__.py", line 312, in _host_allocator
    _lazy_init()
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/cuda/__init__.py", line 120, in _lazy_init
    _check_driver()
  File "/home/yuval/Dev/Python/Shared/anaconda3/lib/python3.6/site-packages/torch/cuda/__init__.py", line 71, in _check_driver
    of the CUDA driver.""".format(str(torch._C._cuda_getDriverVersion())))
AssertionError: 
The NVIDIA driver on your system is too old (found version 9000).
Please update your GPU driver by downloading and installing a new
version from the URL: http://www.nvidia.com/Download/index.aspx
Alternatively, go to: http://pytorch.org to install
a PyTorch version that has been compiled with your version
of the CUDA driver.

My system configuration:
Ubuntu 16.04
Geforce GTX 1080 TI.
**Latest Nvidia driver (384.111)**
CUDA 9.1
CUDNN 7.0
Pytorch 0.3.0 built from source.
Python 3.6.4 :: Anaconda custom (64-bit)

Please help! :(

**Comment by ailzhang** [comment]
Are you able to do `import torch` inside your python terminal?
Probably unrelated: worth trying to remove the ~/.nv if it exists.

**Comment by ngimel** [comment]
You need 387 driver for 9.1 toolkit, 384 corresponds to 9.0 (as the error message tells you)

**Comment by YuvalNirkin** [comment]
@ngimel You are absolutely right. The latest NVIDIA driver offered on NVIDIA's site does not support CUDA 9. Installing the driver from the CUDA 9.1 toolkit solved the problem.

Thanks! :)

**Comment by jramapuram** [comment]
@ngimel : thanks! was wondering why I couldn't compile 0.4a on manjaro. This is it!

**Comment by BEpresent** [comment]
@jramapuram You were able  to install it on Manjaro?

My system says I have CUDA 9.0

```
nvcc --version

nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Sep__1_21:08:03_CDT_2017
Cuda compilation tools, release 9.0, V9.0.176

```

However, explicitly telling it to install the version compiled for 9.0, it ignores it and wants to install the version for cuda91 

```
sudo conda install pytorch torchvision cuda90 -c pytorch

    pytorch:     0.3.1-py36_cuda9.1.85_cudnn7.0.5_2 pytorch [cuda91]
    torchvision: 0.2.0-py36h17b6947_1               pytorch

```
I downgraded Cuda from 9.1 to 9.0 according to https://forum.manjaro.org/t/cuda-9-1-not-compatible-with-nvidia-384/37461/18, but it did not seem to help

I guess this is more a Manjaro error than Pytorch.

**Comment by soumith** [comment]
@BEpresent `conda uninstall -y cuda91; conda install pytorch torchvision cuda90 -c pytorch`

**Comment by jramapuram** [comment]
Yup this. I downgraded cuda to 9.0 and everything works. Check your LD_LIBRARY_PATH and make sure you have conda before /opt/cuda if you dont want to downgrade your system version of CUDA

**Comment by BEpresent** [comment]
This works, thanks!

Then the following error occurred:

`RuntimeError: cuda runtime error (48) : no kernel image is available for execution on the device...`

So just the Cuda 8 version of Pytorch was tried, however, tensorflow-gpu (installed via conda) did not let me install it. The solution was to  `conda uninstall tensorflow-gpu`, then I could install Pytorch (with Cuda 8). Thanks

**Comment by soumith** [comment]
@BEpresent what was the exact command used to install `tensorflow-gpu`? I"ll try reproduce this on my end.

**Comment by BEpresent** [comment]
I installed it some time ago, I  via `conda install tensorflow-gpu`,  but I got a 'Permission denied' error even with `sudo conda uninstall`. In parallel, tensorflow-gpu was also installed with pip3 install, which might have screwed things up. 

```

ERROR conda.core.link:_execute(481): An error occurred while uninstalling package 'defaults::tensorflow-gpu-base-1.4.1-py36ha7e2fe3_0'.
PermissionError(13, 'Permission denied')
```

**Comment by themightyoarfish** [comment]
Same problem with a fresh clone of current master (de099564e32400a593071d7aa4b9a99bb3fe6f0f). As it stands, I cannot compile pytorch on Ubuntu 16.04 with CUDA 8.0 and Nvidia Driver Version: 384.130, which used to work flawlessly.

**Comment by themightyoarfish** [comment]
Same with CUDA 9.0-

**Comment by mrgloom** [comment]
I'm installing pytorch under virtualenv:
```
pip3 install https://download.pytorch.org/whl/cu100/torch-1.0.1.post2-cp35-cp35m-linux_x86_64.whl
pip3 install torchvision
```

I have cuda 10.1 installed:
```
nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Fri_Feb__8_19:08:17_PST_2019
Cuda compilation tools, release 10.1, V10.1.105
```

But for some reason:
```
import torch;torch._C._cuda_getDriverVersion()
9010
```

So I get:
```
AssertionError:
The NVIDIA driver on your system is too old (found version 9010).
Please update your GPU driver by downloading and installing a new
version from the URL: http://www.nvidia.com/Download/index.aspx
Alternatively, go to: https://pytorch.org to install
a PyTorch version that has been compiled with your version
of the CUDA driver.
```

Under /user/local I have:
```
cuda -> /usr/local/cuda-10.1/
drwxr-xr-x 17 root   root    4096 Mar 24 00:14 cuda-10.1
drwxr-xr-x 15 root   root    4096 Nov 29 23:03 cuda-9.0
drwxr-xr-x 15 root   root    4096 Jan 10  2018 cuda-9.1
```

**Comment by soumith** [comment]
@mrgloom your issue is that you have a driver that's only sufficient for CUDA9.1, but you are trying to use CUDA10. Update your driver

**Comment by rderua** [comment]
How to find a table with the equilant to the format 9020 to the name of the GPU? Mine is Geforce 1080 TI x2, and i have that issue too!

**Comment by soumith** [comment]
9020 means it supports CUDA 9.2 but not CUDA10

**Comment by rderua** [comment]
I tried the CUDA9.1 Toolkit and i get an error in the driver import with Anaconda, Windows 10, Geforce x2 1080 ti, importing the driver properties with Pycuda 2018 1.1
ImportError                               Traceback (most recent call last)
<ipython-input-2-ae1f9f7bb38a> in <module>
      1 
----> 2 import pycuda.driver as cuda

C:\ProgramData\Anaconda3\lib\site-packages\pycuda\driver.py in <module>
      3 import six
      4 try:
----> 5     from pycuda._driver import *  # noqa
      6 except ImportError as e:
      7     if "_v2" in str(e):

ImportError: DLL load failed: The specified module could not be found.

**Comment by mrgloom** [comment]
@rderua Take a look at Table 1. CUDA Toolkit and Compatible Driver Versions
https://docs.nvidia.com/deploy/cuda-compatibility/index.html
You can check Driver Version in `nvidia-smi`.

But I wonder does drivers have backward compatibility, i.e. can I have multiply CUDA versions installed in Ubuntu while using fresh driver?

**Comment by rderua** [comment]
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 397.93                 Driver Version: 388.19                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108... WDDM  | 00170000:00:00.0  On |                  N/A |
| 23%   36C    P8    13W / ERR! |   1361MiB / 11264MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108... WDDM  | 00650000:00:00.0 Off |                  N/A |
| 23%   36C    P8    13W / ERR! |   1361MiB / 11264MiB |      2%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0                    ERROR: Unknown Error                                |
|    1                    ERROR: Unknown Error                                |
+-----------------------------------------------------------------------------+


I got this running nvidia-smi,  Unknown Error. Thats why i cant install pycuda?
Which pycuda version suit my os config

**Comment by ashiqimran** [comment]
conda install pytorch=0.4.1 cuda90 -c pytorch 

This command worked for me

**Comment by fhamborg** [comment]
I'm having a similar issue and if anyone knows would highly appreciate help on which pytorch and cudatoolkit version I should install?

I got:
```
>>> torch.cuda.is_available()
False
>>> torch._C._cuda_getDriverVersion()
9020
```

`nvidia-smi` shows I have driver version `396.44`, `nvcc -V` shows `Cuda compilation tools, release 9.0, V9.0.176`. I cannot upgrade the nvidia driver or cuda compiler, since the machine is a University-owned computing cluster, but up until recently, everything worked well, so my current guess is that somehow the pytorch installation got scrambled. I tried various version of pytorch with the cuda python package (which seems to be kind of the old version of the codatoolkit, because I did find cuda only in old tutorials, but more recent sites all mention only codatoolkit, which is also the one you can find on https://pytorch.org/get-started/locally/). None of the version I tried worked. Any ideas?

**Comment by soumith** [comment]
@fhamborg you can either click the CUDA `9.2` button the the get-started selector, or you can upgrade your nvidia driver. Either of those is fine.

**Comment by fhamborg** [comment]
Using the CUDA 9.2 button did not work, e.g.:
![image](https://user-images.githubusercontent.com/18700166/68502013-ccd6cf80-025f-11ea-9d55-e8aa31e105d3.png)

or more specifically, the installation using `conda install pytorch torchvision cudatoolkit=9.2 -c pytorch` is successful, but also leads to cuda not being available in torch.

**Comment by christegho** [comment]
I am having a similar issue as @fhamborg. Although I have CUDA 10.1. Were you able to resolve this issue by any chance?

**Comment by fhamborg** [comment]
Unfortunately not really; I avoided the issue by switching to another machine...

**Comment by soumith** [comment]
@fhamborg @christegho what you both say is pretty strange, I cannot reproduce it, but I am thinking out loud what it could be. Do you by any chance have any previous pytorch installed? Also, can you say what is the output of `print(torch.__config__.show())` and `print(torch.version.cuda)`

**Comment by christegho** [comment]
I ended up resolving the issue by upgrading to CUDA 10.1.

**Comment by shividhar** [comment]
I had the same problem, but I couldn't upgrade the CUDA version because I was working on a cluster where I didn't have root permissions.

The fix is to use the PyTorch package which supports the specific CUDA version. The default package currently supports CUDA 10.1.

[List of PyTorch packages (for Pip)](https://download.pytorch.org/whl/torch_stable.html)
- Sample command to download PyTorch with CUDA 10.0 support 
`pip3 install torch==1.3.1+cu100 torchvision==0.4.2+cu100 -f https://download.pytorch.org/whl/torch_stable.html`

[List of PyTorch packages (for Anaconda)](https://anaconda.org/pytorch/pytorch/files?version=1.3.1)

Most of this information is listed in the [official documentiation](https://pytorch.org/get-started/previous-versions).

**Comment by ajay-sreeram** [comment]
I am having

NVIDIA-SMI <b>390.30</b>                 Driver Version: <b>390.30</b>
Cuda compilation tools, release <b>9.1, V9.1.85</b>
torch._C._cuda_getDriverVersion(): <b>9010</b>

but torch.cuda.current_device() throws error:
`The NVIDIA driver on your system is too old (found version 9010). `

someone please help me with this

**Comment by mossadhelali** [comment]
I had a similar situation to @shividhar, I couldn't update the driver because I didn't have root access. 
The cuda version installed was 10.0. When I installed flair, it updated pytorch because it requires version 1.1+. The problem was the automatic update installed cuda version 10.1. 
So I ran: `pip install torch==1.4.0+cu100 torchvision==0.5.0+cu100 -f https://download.pytorch.org/whl/torch_stable.html` and it worked.

tl;dr: make sure the cuda version matches exactly

**Comment by mikeyEcology** [comment]
@mossadhelali 's solution worked for me when I was getting this error: 
`The NVIDIA driver on your system is too old (found version 10010)`
and I can't update my NVIDIA driver because I'm not an administrator on my HPC, I ran
```
pip install torch==1.4.0+cu100 torchvision==0.5.0+cu100 -f https://download.pytorch.org/whl/torch_stable.html
```
And it's working now.

**Comment by asjt2020** [comment]
I have also same error, i am not filmier with this..
can anyone help me? 

========================= Video dataset #images = 1 =========
======================================  DIW NETWORK TRAIN FROM Ours_Bilinear=======================
===================Loading Pretrained Model OURS ===================================
./checkpoints/test_local\best_depth_Ours_Bilinear_inc_3_net_G.pth
Traceback (most recent call last):
  File "2dtodepth.py", line 32, in <module>
    model = pix2pix_model.Pix2PixModel(opt)
  File "C:\2dtodepth\prog\models\pix2pix_model.py", line 114, in __init__
    new_model.cuda(), device_ids=range(torch.cuda.device_count()))
  File "C:\Program Files\Python37\lib\site-packages\torch\nn\modules\module.py", line 265, in cuda
    return self._apply(lambda t: t.cuda(device))
  File "C:\Program Files\Python37\lib\site-packages\torch\nn\modules\module.py", line 193, in _apply
    module._apply(fn)
  File "C:\Program Files\Python37\lib\site-packages\torch\nn\modules\module.py", line 193, in _apply
    module._apply(fn)
  File "C:\Program Files\Python37\lib\site-packages\torch\nn\modules\module.py", line 199, in _apply
    param.data = fn(param.data)
  File "C:\Program Files\Python37\lib\site-packages\torch\nn\modules\module.py", line 265, in <lambda>
    return self._apply(lambda t: t.cuda(device))
  File "C:\Program Files\Python37\lib\site-packages\torch\cuda\__init__.py", line 162, in _lazy_init
    _check_driver()
  File "C:\Program Files\Python37\lib\site-packages\torch\cuda\__init__.py", line 82, in _check_driver
    http://www.nvidia.com/Download/index.aspx""")
AssertionError:
Found no NVIDIA driver on your system. Please check that you
have an NVIDIA GPU and installed a driver from
http://www.nvidia.com/Download/index.aspx

C:\2dtodepth\prog>pause
Press any key to continue . . .

**Comment by SunLeL** [comment]
I also got this issue, I think it's because CUDA's version doesn't match pytorch's version. The following version is available to me：
cuda9.0 or cuda10.0
pytorch0.4.0(torch==0.4.0)

**Comment by vikash0837** [comment]
I solved it by downgrading the pytorch. That is less cumbersome compared to updating the driver. 
I followed [https://pytorch.org/get-started/previous-versions/] for compatible pytorch version with cuda toolkit
